pkg/healthcheck/healthcheck.go

```diff
@@ -828,6 +829,20 @@ func (hc *HealthChecker) allCategories() []*Category {
 						return hc.checkPodSecurityPolicies(ctx, true)
 					},
 				},
+				{
+					description: "`clusterNetworks` config contains all node podCIDRs",
+					hintAnchor:  "l5d-cluster-networks-cidr",
+					check: func(ctx context.Context) error {
+						// We set `hc.linkerdConfig` so that we dont rely on the "l5d-existence-linkerd-config" check
+						// to use the clusterNetworks value, since `linkerd check config` will skip that check.
+						_, l5dValues, err := FetchCurrentConfiguration(ctx, hc.kubeAPI, hc.ControlPlaneNamespace)
+						if err != nil {
+							return err
+						}
+						hc.linkerdConfig = l5dValues
```

Comment:
You can replace this block with a simpler call to `hc.InitializeLinkerdGlobalConfig(ctx)` which by itself will call `FetchCurrentConfiguration()` with the appropriate args, and it will also initialize `hc.linkerdConfig`.